msprime/ancestry.py

```diff
@@ -364,8 +367,14 @@ def simulator_factory(
             raise ValueError("Cannot provide non-positive sequence length")
         if the_rate < 0:
             raise ValueError("Cannot provide negative recombination rate")
+        if discrete_coordinates and length != math.floor(length):
+            raise ValueError(
+                "Cannot specify continuous sequence length when discrete_coordinates "
```

Comment:
maybe instead of "continuous" say "non-integer"

Reply:
done.